pkg/ingress/controller/apisix_upstream.go

```diff
@@ -112,30 +112,46 @@ func (c *ApisixUpstreamController) processNextWorkItem() bool {
 	return true
 }
 
-func (c *ApisixUpstreamController) syncHandler(key string) error {
-	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+func (c *ApisixUpstreamController) syncHandler(sqo *UpstreamQueueObj) error {
+	namespace, name, err := cache.SplitMetaNamespaceKey(sqo.Key)
 	if err != nil {
-		log.Errorf("invalid resource key: %s", key)
-		return fmt.Errorf("invalid resource key: %s", key)
+		log.Errorf("invalid resource key: %s", sqo.Key)
+		return fmt.Errorf("invalid resource key: %s", sqo.Key)
 	}
-
-	apisixUpstreamYaml, err := c.apisixUpstreamList.ApisixUpstreams(namespace).Get(name)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			log.Infof("apisixUpstream %s is removed", key)
+	apisixUpstreamYaml := sqo.OldObj
+	if sqo.Ope == DELETE {
+		apisixIngressUpstream, _ := c.apisixUpstreamList.ApisixUpstreams(namespace).Get(name)
+		if apisixIngressUpstream != nil && apisixIngressUpstream.ResourceVersion > sqo.OldObj.ResourceVersion {
+			log.Warnf("Upstram %s has been covered when retry", sqo.Key)
 			return nil
 		}
-		runtime.HandleError(fmt.Errorf("failed to list apisixUpstream %s/%s", key, err.Error()))
-		return err
+	} else {
+		apisixUpstreamYaml, err = c.apisixUpstreamList.ApisixUpstreams(namespace).Get(name)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				log.Infof("apisixUpstraem %s is removed", sqo.Key)
+				return nil
+			}
+			runtime.HandleError(fmt.Errorf("failed to list apisixService %s/%s", sqo.Key, err.Error()))
```

Comment:
Type: here we are listing `ApisixUpstream`.

Reply:
Done